src/threadlist.cpp

```diff
@@ -686,19 +687,59 @@ ThreadList::waitForAllRestored(Thread *thread)
     sem_wait(&semWaitForCkptThreadSignal);
   }
   Thread_RestoreSigState(thread);
+
+  if (thread == motherofall) {
+    /* If DMTCP_RESTART_PAUSE==4, sleep 15 seconds to allow gdb attach.*/
+    char * pause_param = getenv("DMTCP_RESTART_PAUSE");
+    if (pause_param == NULL) {
+      pause_param = getenv("MTCP_RESTART_PAUSE");
+    }
+    if (pause_param != NULL && pause_param[0] == '4' &&
+        pause_param[1] == '\0') {
+#ifdef HAS_PR_SET_PTRACER
+      prctl(PR_SET_PTRACER, PR_SET_PTRACER_ANY, 0, 0, 0); // For: gdb attach
+#endif // ifdef HAS_PR_SET_PTRACER
+      struct timespec delay = { 15, 0 }; /* 15 seconds */
```

Comment:
Should we move this entire block in the ifdef? It makes no sense to sleep if HAS_PR_SET_PTRACER is not set. We won't be able to attach gdb anyway.